Issues 1966

raftaar1191 opened the pull request:
## Description
PR for featured #1966 

## How Has This Been Tested?
Manually Tested

## Video Link (jpeg or gifs if applicable):
http://screencast-o-matic.com/watch/cbjFrmlM1X

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

raftaar1191 commented:
@ravinderk  @DevinWalker 

Please review this PR and merge 

For more information on the PR please link on the below video link http://screencast-o-matic.com/watch/cbjFrmlM1X

ravinderk commented:
@raftaar1191 please test your pr with lowest and highest supported version of PHP when you ask to review PR.
Getting this error `PHP Fatal error:  Can't use method return value in write context in /app/public/wp-content/plugins/Give/includes/misc-functions.php on line 1527` on PHP`5.4`

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -2190,3 +2196,72 @@ function get_url_params() {
+function give_on_donation_import_ajax() {
+    var $form = jQuery('form.tools-setting-page-import');
+
+    var progress = jQuery( $form ).find( '.give-progress' );
```
@raftaar1191 No need to wrapper `$form` again with`jQuery`. It is already a `jQuery` object

raftaar1191 replied:
Solved

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -2190,3 +2196,72 @@ function get_url_params() {
+    var $form = jQuery('form.tools-setting-page-import');
+
+    var progress = jQuery( $form ).find( '.give-progress' );
+    var total_ajax = jQuery( progress ).data( 'total_ajax' );
```
@raftaar1191 Do not need to add `var` for each variable, use javascript multi-line variable declaration structure

raftaar1191 replied:
Solved thanks

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -2190,3 +2196,72 @@ function get_url_params() {
+        type: 'POST',
+        url: ajaxurl,
+        data: {
+            action: give_vars.give_donation_import,
```
@raftaar1191 Why we need dynamic action for this process? Can we hardcode this?

raftaar1191 replied:
Yes we can there is not differences

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -2190,3 +2196,72 @@ function get_url_params() {
+        },
+        dataType: 'json',
+        success: function ( response ) {
+            console.log( response );
```
@raftaar1191 Please remove `console.log` after testing

raftaar1191 replied:
Remove thanks

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -388,11 +388,120 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
 add_action( 'admin_bar_menu', '_give_show_test_mode_notice_in_admin_bar', 1000, 1 );
 
+
+function give_import_page_link_callback() {
```
@raftaar1191 Add code doc block to this function

raftaar1191 replied:
Done thanks

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -388,11 +388,120 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+function give_import_page_link_callback() {
+	?>
+    <a href="<?php echo esc_url( give_import_page_url() ); ?>"
+       class="page-import-action page-title-action"><?php esc_html_e( 'Import Donations', 'give-manual-donations' ); ?></a>
```
@raftaar1191 Fix text domain

raftaar1191 replied:
Done thanks

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -388,11 +388,120 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+	<?php
+
+	// Check if view donation single page only.
+	if ( ! empty( $_REQUEST['view'] ) && 'view-payment-details' === (string) sanitize_text_field( $_REQUEST['view'] ) && 'give-payment-history' === (string) sanitize_text_field( $_REQUEST['page'] ) ) {
```
@raftaar1191 Use `give_clean`

raftaar1191 replied:
Done thanks

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -388,11 +388,120 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+	// Parent key id.
+	$main_key = maybe_unserialize( $main_key );
+
+	$current    = (int) sanitize_text_field( $_REQUEST['current'] );
```
@raftaar1191 use `absint` instead

raftaar1191 replied:
Done thanks

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -388,11 +388,120 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+	if ( $next == false ) {
+		$json_data = array(
+			'success' => true,
+			'message' => __( 'All donation uploaded successfully!', 'give-manual-donations' ),
```
@raftaar1191 Fix text domain

raftaar1191 replied:
Done thanks

ravinderk commented on includes/admin/tools/views/html-admin-page-imports.php:
```diff
@@ -0,0 +1,70 @@
+/**
+ * Fires before displaying the import div tools.
+ *
+ * @since 1.5
```
@raftaar1191 Why @since tag set to `1.5`

raftaar1191 replied:
Done thanks

ravinderk commented on includes/process-donation.php:
```diff
@@ -895,8 +897,8 @@ function give_get_donation_form_user( $valid_data = array() ) {
 	$user['address']['state']   = ! empty( $_POST['card_state'] ) ? sanitize_text_field( $_POST['card_state'] ) : false;
-	$user['address']['country'] = ! empty( $_POST['billing_country'] ) ? sanitize_text_field( $_POST['billing_country'] ) : false;
 	$user['address']['zip']     = ! empty( $_POST['card_zip'] ) ? sanitize_text_field( $_POST['card_zip'] ) : false;
+	$user['address']['country'] = ! empty( $_POST['billing_country'] ) ? ( ( $country_code = array_search( sanitize_text_field( $_POST['card_zip'] ), give_get_country_list() ) ) ? $country_code : sanitize_text_field( $_POST['card_zip'] ) ) : false;
```
@raftaar1191 use `give_clean`

raftaar1191 replied:
Done thanks

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -2190,3 +2196,72 @@ function get_url_params() {
+            }
+        },
+        error: function () {
+            alert( give_vars.error_message );
```
@raftaar1191 are you sure this will show an error message?

raftaar1191 replied:
yes it's working

DevinWalker commented:
@raftaar1191 change request: 

![2017-08-22_13-59-45](https://user-images.githubusercontent.com/1571635/29587591-44a38890-8743-11e7-9871-facd80b72f7f.png)

DevinWalker commented:
@raftaar1191 a few more items:

## 1. WP User Creation

Under what criteria is a WP user created? Some people may or may not want to create WP users for their donors. I'd like to discuss the best approach here for:

1. Creating _new_ WP users for new imported donors - I was thinking maybe adding an option during the import process like "Create WP users for new donors?"  
2.  Associating _existing_ WP users for new imported donors (possibly by providing an existing WP User ID via a column in the CSV).

## 2. Repeated Imports and Duplicates

I noticed that I am able to keep importing over and over again. It appears that the donors and forms are not duplicate, which is good. However, the _donations_ are created. Is this intentional? If an imported donation happened on the _same_ date and time it shouldn't be imported. That's a duplicate. 

## 3. Final Import "Report"

When you are done importing the breakdown isn't very descriptive. There is a lot more information that could be listed:

1. Number of duplicate donors, forms, wp users, and donations detected and skipped
2. The number of donors, forms, wp users, and donations created

Please update so that the final report is more descriptive so admins have a better idea what imported, what was skipped, and why.
 
![2017-08-22_14-26-16](https://user-images.githubusercontent.com/1571635/29588435-5dff3516-8746-11e7-912b-a712cdbd994e.png)

## 4. Test vs Live Donations

Currently there is no way the I can see to specify whether an imported donation has been made in LIVE or TEST mode. I have added a column to the Google Doc spread sheet which includes a new "Test Mode" column that is boolean.  If this column does _not_ exist within the import the admin should still be able to specify whether the import is should be in test or live mode.

![2017-08-22_14-33-10](https://user-images.githubusercontent.com/1571635/29588588-fe8ab230-8746-11e7-9923-7791f602fa96.png)

## 5. New Donors Donations Count

This appears to be an issue with breaking the cache or an improper import that needs fixing:

![2017-08-22_14-21-59](https://user-images.githubusercontent.com/1571635/29588176-6581674c-8745-11e7-9d59-dea8f28fd4b2.png)

## 6. Remove the "Imported" badge and Add "Imported on ..." Note

Please remove this badge here:

![2017-08-22_14-23-35](https://user-images.githubusercontent.com/1571635/29588805-b9e031e0-8747-11e7-8faa-346968ee2bde.png)

And in its place add a payment note: "This donation was imported by admin@website.com" that should be the _first_ note for the payment:

![2017-08-22_14-38-33](https://user-images.githubusercontent.com/1571635/29588798-b58ded08-8747-11e7-9ec8-ef5df623adac.png)

Also, for any _donor_ that is imported there also need to be a note:

![2017-08-22_14-40-43](https://user-images.githubusercontent.com/1571635/29588920-177c98fc-8748-11e7-8b9a-ec43f5874262.png)

## 7. Empty Note Created on Import

I noticed that the imported is creating a note even if the note is blank:

![2017-08-22_14-24-12](https://user-images.githubusercontent.com/1571635/29588684-521baad0-8747-11e7-8547-3fc831fdbeff.png)

## 8. CSS Issue with "Import Donations" button

Deactivate the "Manual Donations" add-on and then see the CSS issue here:

![2017-08-22_14-44-14](https://user-images.githubusercontent.com/1571635/29589044-961593c6-8748-11e7-9951-6a6b7c655889.png)

## 9. Delete Imported Data Tool Not Deleting All Data

I know our users are going to try this out and some will want to remove the imported data. The tool you've built seems to delete _some_ of the data but some still remains after running:

![2017-08-22_14-54-26](https://user-images.githubusercontent.com/1571635/29589373-dce7afea-8749-11e7-9087-f9af989896b0.png)

## 10. Typos to Fix

![2017-08-22_14-49-36](https://user-images.githubusercontent.com/1571635/29589281-8dc477a4-8749-11e7-9b2a-089dd248d8d4.png)

![2017-08-22_14-50-52](https://user-images.githubusercontent.com/1571635/29589282-90a9f750-8749-11e7-80f2-19eb1d32a8fd.png)

![2017-08-22_14-53-19](https://user-images.githubusercontent.com/1571635/29589326-bb8688e4-8749-11e7-81af-f4dbb10da383.png)

![2017-08-22_14-46-43 copy](https://user-images.githubusercontent.com/1571635/29589161-0556c05c-8749-11e7-9048-32acdc6ace02.png)

![2017-08-22_14-46-43](https://user-images.githubusercontent.com/1571635/29589146-f13e0026-8748-11e7-98da-e2adc562aa1e.png)

---

Please update the imported with the aforementioned issues. Ping me if you have any questions. Thanks!

DevinWalker commented:
How's it going on this one @raftaar1191 can I get an update?

raftaar1191 commented:
Solved all the Changes request from @ravinderk 


Related to all your change request 
1. WP User Creation
In the CSV there is an option to enter Donor ID and Donor Email id both. If donor id is not empty then it will search for that ID in the user list. If found then all the details will be fetched from that user id or if that id does not exist then it will search for the email if that email is already registered then all the details from that email will be fetch and if that email is not being registered then it will register a new user with that email followed by first name and last name.


2. Repeated Imports and Duplicates
Solved, Now where a new donation is imported it will first check if this donation already exists or not. It will first check with the same Form id, price, author, and date as well. If all the match then it assumes to be the same donation and if not then 

3. Final Import "Report"
Working on it

4. Test vs Live Donations
Now there is a column in the CSV to mention that the donation is ( test or live ) donation. The value should in a boolean value. If in the column no found then there is an option on the first stepto choose the type of donation 

5. New Donors Donations Count
Working on it

6. Remove the "Imported" badge and Add "Imported on ..." Note
Done, Added in donation single page as well in donor notes section 

7. Empty Note Created on Import
Solved 

8. CSS Issue with "Import Donations" button
Solved

9. Delete Imported Data Tool Not Deleting All Data
Working on it

10. Typos to Fix
Fixed


11. Media Uploader Modal
Working on it

raftaar1191 commented:
@DevinWalker  
Related to you comment point number 
**11. Media Uploader Modal**
->It will be better to used ID in place of URL

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -1340,11 +1346,18 @@ jQuery.noConflict();
 							button  : {text: give_vars.metabox_fields.file.button_title},
-							multiple: false
+							multiple: false,
+                            contentUserSetting :false,
```
@deepak can you explain this var?

raftaar1191 replied:
It is being used to show the uploader tab when **add media button** is click

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -391,9 +391,143 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+function give_import_page_link_callback() {
+	?>
+    <a href="<?php echo esc_url( give_import_page_url() ); ?>"
+       class="page-import-action page-title-action"><?php esc_html_e( 'Import Donations', 'give' ); ?></a>
```
@raftaar1191 unnecessary `esc_html`. Remove it.

raftaar1191 replied:
@ravinderk 
I was reading this article that why I used it. Please have a look into it
https://vip.wordpress.com/documentation/validating-sanitizing-escaping/#always-escape-late

DevinWalker replied:
This is OK @raftaar1191

ravinderk replied:
@raftaar1191 In case string is hard coded, not user defined

raftaar1191 replied:
Done

ravinderk commented on includes/process-donation.php:
```diff
@@ -815,8 +815,10 @@ function give_register_and_login_new_user( $user_data = array() ) {
 
-	// Login new user.
-	give_log_user_in( $user_id, $user_data['user_login'], $user_data['user_pass'] );
+	if ( true === (bool) apply_filters( 'give_log_user_in_on_register', true ) ) {
```
@raftaar1191 Add documentation to this filter

raftaar1191 replied:
Done

ravinderk commented on includes/process-donation.php:
```diff
@@ -891,12 +893,12 @@ function give_get_donation_form_user( $valid_data = array() ) {
+	$user['address']['city']    = ! empty( $_POST['card_city'] ) ? give_clean( $_POST['card_city'] ) : false;
+	$user['address']['state']   = ! empty( $_POST['card_state'] ) ? give_clean( $_POST['card_state'] ) : false;
+	$user['address']['zip']     = ! empty( $_POST['card_zip'] ) ? give_clean( $_POST['card_zip'] ) : false;
+	$user['address']['country'] = ! empty( $_POST['billing_country'] ) ? ( ( $country_code = array_search( give_clean( $_POST['card_zip'] ), give_get_country_list() ) ) ? $country_code : give_clean( $_POST['card_zip'] ) ) : false;
```
@raftaar1191 Can you explain this

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+					if ( ! empty( $report['duplicate_donor'] ) ) {
+						?>
+                        <p>
+							<?php echo esc_html( wp_sprintf( '%s duplicate donors detected', $report['duplicate_donor'] ), 'give' ); ?>
```
@raftaar1191 Please check this implementation. Fix them all.
<img width="946" alt="screen shot 2017-08-29 at 2 28 56 pm" src="https://user-images.githubusercontent.com/1784821/29813066-6b0e49d8-8cc6-11e7-944f-f134155c5bd4.png">

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+                    <h2>
+						<?php
+						if ( $success ) {
+							echo sprintf( __( 'Import complete! %s donations imported', 'give' ), '<strong>' . $total . '</strong>' );
```
@raftaar1191 Use double quotes.
```
'<strong>' . $total . '</strong>'  --> "<strong>{$total}</strong>"
```

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+                        </p>
+						<?php
+					}
+					?>
```
@raftaar1191 Collect import report in an array and then print them once

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+		}
+
+		/**
+		 * Will return true if importing can be started or not else false.
```
@raftaar1191 Please add missing `@since` tag atleast

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+			?>
+            <tr valign="top" class="give-import-dropdown">
+                <th colspan="2">
+                    <h2 id="give-import-title"><?php esc_html_e( 'Map CSV fields to donations', 'give' ) ?></h2>
```
@raftaar1191 remove unnecessary `esc_html` from hard coded strings

raftaar1191 replied:
@ravinderk 

I was reading this article that why I used it. Please have a look into it
https://vip.wordpress.com/documentation/validating-sanitizing-escaping/#always-escape-late

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+			return $total;
+		}
+
+		static function get_importer( $file_id, $index = 0, $delimiter = ',' ) {
```
@raftaar1191 Please add documentation of function if newly created

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+			$step = Give_Settings_Import::get_step();
+			?>
+            <ol class="give-progress-steps">
+                <li class="<?php echo( 1 === $step ? 'active' : '' ); ?>">Upload CSV file</li>
```
@raftaar1191 Is there any reason that these strings are not internationalize

raftaar1191 replied:
Done

ravinderk commented on includes/admin/tools/class-settings-import.php:
```diff
@@ -0,0 +1,681 @@
+		 * @return int $step on which step doest the import is on.
+		 */
+		static function get_step() {
+			$step    = (int) ( isset( $_REQUEST['step'] ) ? give_clean( $_REQUEST['step'] ) : false );
```
@raftaar1191 if accept the only int then why we set the default value to `false`. This should be `0`, right?

raftaar1191 replied:
@ravinderk did not get you, Can you please explain it again

raftaar1191 replied:
Got it. Fixed!

ravinderk commented on includes/admin/tools/data/class-give-tools-delete-import-donors.php:
```diff
@@ -0,0 +1,442 @@
+	/**
+	 * Given a key, get the information from the Database Directly
+	 *
+	 * @since  1.8.12
```
@raftaar1191 is this wrong `@since` tag, please update them if any

raftaar1191 replied:
Done

DevinWalker commented:
@raftaar1191 Please see my response:

> WP User Creation
> In the CSV there is an option to enter Donor ID and Donor Email id both. If donor id is not empty then it will search for that ID in the user list. If found then all the details will be fetched from that user id or if that id does not exist then it will search for the email if that email is already registered then all the details from that email will be fetch and if that email is not being registered then it will register a new user with that email followed by first name and last name.

There should be _two_ separate columns in the CSV. Donor ID corresponds to the ID of the _Give_ donor. User ID corresponds to the ID of the _WordPress_ user. If user ID is provided and no donor ID, the donor is created and attached to that WP user. If a Donor ID exists and not a WP User, then the donor is first searched for, and if found then the details are used for that existing donor. If the donor is not found then the donor is created. Like wise with WP user. 

It would be _really_ great if the importer could log and output specific exceptions like above. For instance 

_"The importer could not find a donor with ID #234. New donor # 1345 was created."_

We can add this in a future version. Let's first focus on making it work right.

> Repeated Imports and Duplicates
> Solved, Now where a new donation is imported it will first check if this donation already exists or not. It will first check with the same Form id, price, author, and date as well. If all the match then it assumes to be the same donation and if not then

Great!

> Final Import "Report"
> Working on it ( Done 29.08.17 )

Thanks.

> Test vs Live Donations
> Now there is a column in the CSV to mention that the donation is ( test or live ) donation. The value should in a boolean value. If in the column no found then there is an option on the first stepto choose the type of donation

Great

> New Donors Donations Count
> Working on it ( Done 29.08.17 )

👍🏻

> Remove the "Imported" badge and Add "Imported on ..." Note
> Done, Added in donation single page as well in donor notes section
>
> Empty Note Created on Import
> Solved
> 
> CSS Issue with "Import Donations" button
> Solved
> 
> Delete Imported Data Tool Not Deleting All Data
> Working on it ( Done 29.08.17 )
> 
> Typos to Fix
> Fixed

👍🏻

> Media Uploader Modal
> Working on it

@raftaar1191 what's the status on this?

> Related to your comment point number 11
> 11. Media Uploader Modal
> ->It will be better to used ID in place of URL

Ok I guess it's not a big deal.

raftaar1191 commented:
@DevinWalker 

Solved all the issues and also added a column for User ID as well

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -391,9 +391,143 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+add_action( 'give_view_order_details_before', 'give_import_page_link_callback', 11 );
+
+
+function give_donation_import_callback() {
```
@raftaar1191 Add function description

raftaar1191 replied:
Done

ravinderk commented on includes/admin/admin-actions.php:
```diff
@@ -391,9 +391,143 @@ function _give_show_test_mode_notice_in_admin_bar( $wp_admin_bar ) {
+	$import_setting = array();
+	$fields         = isset( $_POST['fields'] ) ? $_POST['fields'] : null;
+
+	parse_str( $fields );
```
@raftaar1191 use `give_clean` to sanitize values

raftaar1191 replied:
Done

DevinWalker commented:
@raftaar1191 we're almost there! Please see the following:

## WP Users are being created without roles
![2017-08-31_13-46-57](https://user-images.githubusercontent.com/1571635/29944630-0f717382-8e53-11e7-80e2-6dbf0275afc6.png)

See these users, without roles, are being created when I run an import. They should have the new Give Donor and WP's subscriber roles.

## Add Option to Delete CSV after import

If you upload a CSV with sensitive information to `wp-content/uploads/` it can be publicly accessible. Please add an option to delete the CSV after import for better security. 

![2017-08-31_13-53-16](https://user-images.githubusercontent.com/1571635/29944879-ed45197a-8e53-11e7-9b73-2f7a4b93ee6e.png)

raftaar1191 commented:
@DevinWalker  working on the above

raftaar1191 commented:
@DevinWalker  Solved and add new featured

ravinderk commented on includes/misc-functions.php:
```diff
@@ -1274,9 +1274,644 @@ function give_set_upgrade_complete( $upgrade_action = '' ) {
+}
+
+
+function give_save_import_donation_to_db( $raw_key, $row_data, $main_key = array(), $import_setting = array() ) {
```
@raftaar1191 Missing function documentation

raftaar1191 replied:
Done

raftaar1191 replied:
Done

ravinderk commented on includes/misc-functions.php:
```diff
@@ -1274,9 +1274,644 @@ function give_set_upgrade_complete( $upgrade_action = '' ) {
+/**
+ * Check if Import donation is duplicate
+ *
+ * Ref: https://xwp.github.io/engineering-best-practices/php/
```
@raftaar1191 Why did you add this

raftaar1191 replied:
The argument that is being passed to WP query will result in better performance of MYSQL

So I thought of putting a referral link from which source did I get this will be useful

ravinderk replied:
Remove this one.

raftaar1191 replied:
Done

ravinderk commented on includes/misc-functions.php:
```diff
@@ -1363,3 +1995,227 @@ function give_array_insert_after( $key, array &$array, $new_key, $new_value ) {
+ * @return mixed Will be an array if $single is false. Will be value of meta data field if $single
+ *  is true.
+ */
+function get_customer_meta( $customer_id, $key = '', $single = false ) {
```
@raftaar1191 User donor instead of customer

raftaar1191 replied:
Done

DevinWalker commented:
## Importer Review 

#### The "Delete CSV after import" should be deleted by default

![2017-09-05_15-09-37](https://user-images.githubusercontent.com/1571635/30085801-192ff33c-924d-11e7-8443-1d546e7c1484.png)

#### Move "Do not import" to the first position of the mapping dropdown

![2017-09-05_15-10-43](https://user-images.githubusercontent.com/1571635/30085811-25f581cc-924d-11e7-89fa-f1e93ca31eda.png)

#### Use `_n()` to output appropriate language for singular usage 

![2017-09-05_15-11-55](https://user-images.githubusercontent.com/1571635/30085834-3df9eb50-924d-11e7-98d0-131ef7f4fe04.png)

#### Change to a button using `.button` WP core class

![2017-09-05_15-13-14](https://user-images.githubusercontent.com/1571635/30085849-4cd84fe0-924d-11e7-9a4d-a366e85f9ebb.png)

#### Fix bug where imported for levels all have default radio checked

![2017-09-05_15-14-33](https://user-images.githubusercontent.com/1571635/30085861-5704908c-924d-11e7-96aa-445e8ca75ff4.png)

#### Imported custom donations don't display "Custom" and created forms have "Custom" disabled incorrect

When a donation doesn't match a level it's marked as "Custom". I noticed these donations missing "Custom" and the form has "Custom Amount" to disabled incorrectly.

![2017-09-05_15-18-37](https://user-images.githubusercontent.com/1571635/30086021-087920d0-924e-11e7-9878-615d6ea2de89.png)

![2017-09-05_15-19-08](https://user-images.githubusercontent.com/1571635/30086026-0bcd8456-924e-11e7-8ef3-ab6186d9ea7c.png)

![2017-09-05_15-21-15](https://user-images.githubusercontent.com/1571635/30086033-0e82f9ec-924e-11e7-908a-8cf1a103a38d.png)

raftaar1191 commented:
@DevinWalker 

Fixed all the error and suggestion that you have added above

DevinWalker commented:
Great job on this @raftaar1191

raftaar1191 commented:
Thanks @DevinWalker
